### PR TITLE
AWS WAF V2 Implementation

### PR DESCRIPTION
**Pull request by michelvocks**
Implements https://github.com/cloudquery/cloudquery/issues/117

### CODE REVIEW

File: resources/waf_v2_rule_groups.go [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+import (
+	"context"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/wafv2"
+	"github.com/cloudquery/cq-provider-aws/client"
+	"github.com/cloudquery/cq-provider-sdk/provider/schema"
+)
+
+func WafV2RuleGroups() *schema.Table {
```
Comment (roneli) [inline resources/waf_v2_rule_groups.go:12]: missing [GetPermissionPolicy ](https://docs.aws.amazon.com/waf/latest/APIReference/API_GetPermissionPolicy.html)for the RuleGroup add this a json policy column

File: resources/waf_v2_managed_rule_groups.go [hunk 2]
```diff
@@ -0,0 +1,68 @@
+package resources
```
Comment (roneli) [inline resources/waf_v2_managed_rule_groups.go:1]: maybe a naming thing, might I suggest wafv2_managed_rule_groups? wdyt?

**Review by roneli — CHANGES_REQUESTED** [review]
Missing a few APIs to add a little more data  to some tables, other than that looks good

File: resources/waf_v2_webacls.go [hunk 3]
```diff
@@ -0,0 +1,425 @@
...
+	"fmt"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/wafv2"
+	"github.com/aws/aws-sdk-go-v2/service/wafv2/types"
+	"github.com/cloudquery/cq-provider-aws/client"
+	"github.com/cloudquery/cq-provider-sdk/provider/schema"
+)
+
+func WafV2Webacls() *schema.Table {
```
[the remark this answers is not in the document]
Comment (michelvocks) [reply resources/waf_v2_webacls.go:15]: Thanks!

File: resources/waf_v2_managed_rule_groups.go (hunk 2 above)
Comment (michelvocks) [reply resources/waf_v2_managed_rule_groups.go:1]: Makes sense! Thanks

File: resources/waf_v2_webacls.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (michelvocks) [reply resources/waf_v2_webacls.go]: Thanks! I couldn't find tags for managed rule groups so I skipped these.

File: resources/waf_v2_rule_groups.go [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+				Type: schema.TypeString,
+			},
+		},
+	}
+}
+
+// ====================================================================================================================
+//                                               Table Resolver Functions
+// ====================================================================================================================
+func fetchWafV2RuleGroups(ctx context.Context, meta schema.ClientMeta, parent *schema.Resource, res chan interface{}) error {
```
[the remark this answers is not in the document]
Comment (michelvocks) [reply resources/waf_v2_rule_groups.go]: Thanks!

File: resources/waf_v2_managed_rule_groups.go [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+import (
+	"context"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/wafv2"
+	"github.com/cloudquery/cq-provider-aws/client"
+	"github.com/cloudquery/cq-provider-sdk/provider/schema"
+)
+
+func WafV2ManagedRuleGroups() *schema.Table {
```
[the remark this answers is not in the document]
Comment (michelvocks) [reply resources/waf_v2_managed_rule_groups.go]: Thanks!

File: resources/waf_v2_rule_groups.go (hunk 1 above)
Comment (michelvocks) [reply resources/waf_v2_rule_groups.go]: Thanks!

**Review by roneli — APPROVED** [review]
Looks great! that was a big service

**Pull request merged**